python/cuml/dask/common/comms.py

```diff
@@ -219,119 +117,52 @@ def _func_build_handle(sessionId, streams_per_handle):
-        count += 1
-
-    worker_state(sessionId)["ucp_eps"] = eps
-
-
-async def _func_destroy_all(sessionId, comms_p2p):
-    worker_state(sessionId)["nccl"].destroy()
-    del worker_state(sessionId)["nccl"]
-    del worker_state(sessionId)["handle"]
-
-
-def _func_ucp_ports(client, workers):
-    return client.run(_func_ucp_listener_port,
-                      workers=workers)
-
-
-def _func_worker_ranks(workers):
-    """
-    Builds a dictionary of { (worker_address, worker_port) : worker_rank }
-    """
-    return dict(list(zip(workers, range(len(workers)))))
-
-
-class CommsContext:
-
+def worker_state(sessionId=None):
     """
-    A base class to initialize and manage underlying NCCL and UCX
-    comms handles across a Dask cluster. Classes extending CommsContext
-    are responsible for calling `self.init()` to initialize the comms.
-    Classes that extend or use the CommsContext are also responsible for
-    calling `destroy()` to clean up the underlying comms.
-
-    This class is not meant to be thread-safe.
+    Retrieves cuML comms state on local worker for the given
+    sessionId, creating a new session if it does not exist.
+    If no session id is given, returns the state dict for all
+    sessions.
+    :param sessionId:
+    :return:
     """
+    worker = get_worker()
+    if not hasattr(worker, "_raft_comm_state"):
+        worker._raft_comm_state = {}
+    if sessionId is not None and sessionId not in worker._raft_comm_state:
+        # Build state for new session and mark session creation time
+        worker._raft_comm_state[sessionId] = {"ts": time.time()}
 
-    def __init__(self, comms_p2p=False, client=None, streams_per_handle=0):
-        """
-        Construct a new CommsContext instance
-        :param comms_p2p: bool Should p2p comms be initialized?
-        """
-        self.client = client if client is not None else default_client()
-        self.comms_p2p = comms_p2p
-
-        self.streams_per_handle = streams_per_handle
-
-        self.sessionId = uuid.uuid4().bytes
-
-        self.nccl_initialized = False
-        self.ucx_initialized = False
-
-        if comms_p2p and (not is_ucx_enabled() or not has_ucp()):
-            warnings.warn("ucx-py not found. UCP Integration will "
-                          "be disabled.")
-            self.comms_p2p = False
-
-        logger.debug("Initializing comms!")
+    if sessionId is not None:
+        return worker._raft_comm_state[sessionId]
+    return worker._raft_comm_state
 
-    def __del__(self):
-        if self.nccl_initialized or self.ucx_initialized:
-            self.destroy()
 
-    def worker_info(self, workers):
-        """
-        Builds a dictionary of { (worker_address, worker_port) :
-                                (worker_rank, worker_port ) }
-        """
-        ranks = _func_worker_ranks(workers)
-        ports = _func_ucp_ports(self.client, workers) \
-            if self.comms_p2p else None
+class CommsContext(Comms):
```

Comment:
This should be able to be replaced with RAFT’s comms directly. We shouldn’t have to define it in cuml anymore at all.